packages/fiori/src/ShellBar.js

```diff
@@ -464,14 +466,24 @@ class ShellBar extends UI5Element {
 		};
 
 		this._handleResize = async event => {
-			await this._getResponsivePopover();
-			this.overflowPopover.close();
-			this._overflowActions();
+			this._debounce(async () => {
+				await this._getResponsivePopover();
+				this.overflowPopover.close();
+				this._overflowActions();
+			}, HANDLE_RESIZE_DEBOUNCE_RATE);
 		};
 
 		this.i18nBundle = getI18nBundle("@ui5/webcomponents-fiori");
 	}
 
+	_debounce(fn, delay) {
```

Comment:
There is a debounce function in the base project which should be imported and used.
It is located here: ```packages/base/src/util/debounce.js```

Reply:
Yes, but it needs to be changed. We have already discussed that with @ilhan007 

The util debounce function will always execute only the latest function which was passed in the given interval. Currently, there are three different controls, calling it, with the shellbar there will be four controls. If some/all of them (or even different instances of these controls) call debounce at the same time, only the last function will be executed, which is not OK. That's why here we have this._debounceInterval per each instance. 

In my opinion, general utility function should be even more flexible and implemented as part of a different task with clear requirements.